File: src/main/java/com/caracal/camel/routes/models/Customer.java
package com.caracal.camel.routes.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Customer {
    private String name;
    private String surname;
    private int age;
    private Address address;

    public Customer() {
        this.setName("");
        this.setSurname("");
        this.setAge(0);
        this.setAddress(new Address());
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }
}

File: src/main/java/com/caracal/camel/routes/models/Customer.kt
package com.caracal.camel.routes.models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.util.UUID

@JsonIgnoreProperties(ignoreUnknown = true)
data class Customer(
    val name: String = "",
    val surname: String = "",
    val age: Int = 0,
    val address: Address = Address())

data class Address(
    val street: String = "",
    val city: String = "",
    val state: String = "",
    val zip: String = "")

@JsonIgnoreProperties(ignoreUnknown = true)
data class CustomerResponse(
    val id: UUID = UUID.randomUUID(),
    val name: String = "",
    val surname: String = "",
    val message: String = ""
)
